Source/FakeItEasy/Core/AssemblyExtensions.cs

```diff
@@ -5,7 +5,7 @@
 
     internal static class AssemblyExtensions
     {
-        private static readonly string FakeItEasyAssemblyName = Assembly.GetExecutingAssembly().FullName;
+        internal static readonly Assembly FakeItEasyAssembly = Assembly.GetExecutingAssembly();

```

Comment:
Actually, it shouldn't be a property on this type, since this type ought to remain strictly an extension method container.

I think I'd prefer to just leave the field repeated in `TypeCatalogue`. Or alternatively, it could be exposed as a static property on `TypeCatalogue` to be reused by `AssemblyExtensions`.